src/lib/snark_work_lib/metrics.ml

```diff
@@ -69,94 +65,49 @@ let emit_single_metrics ~(single_spec : _ Single_spec.Poly.t)
-                Counter.inc_one snark_work_zkapp_base_submissions ;
-                Counter.inc zkapp_transaction_length
-                  (Float.of_int parties_count) ;
-                Counter.inc zkapp_proof_updates
-                  (Float.of_int proof_parties_count))) ;
-            (`Zkapp_command, parties_count, proof_parties_count)
-        | Mina_transaction.Transaction.Command
-            (Mina_base.User_command.Signed_command _) ->
-            Mina_metrics.(
-              Counter.inc Cryptography.snark_work_non_zkapp_base_time_sec
-                (Time.Span.to_sec elapsed)) ;
-            (`Signed_command, 1, 0)
-        | Mina_transaction.Transaction.Coinbase _ ->
-            Mina_metrics.(
-              Counter.inc Cryptography.snark_work_non_zkapp_base_time_sec
-                (Time.Span.to_sec elapsed)) ;
-            (`Coinbase, 1, 0)
-        | Mina_transaction.Transaction.Fee_transfer _ ->
-            Mina_metrics.(
-              Counter.inc Cryptography.snark_work_non_zkapp_base_time_sec
-                (Time.Span.to_sec elapsed)) ;
-            (`Fee_transfer, 1, 0)
+        | Mina_transaction.Transaction.Command (Zkapp_command _) ->
+            `Zkapp_command
+        | Command (Signed_command _) ->
+            `Signed_command
+        | Coinbase _ ->
+            `Coinbase
+        | Fee_transfer _ ->
+            `Fee_transfer
       in
-      Base_snark_generated
-        { transaction_type
-        ; elapsed
-        ; zkapp_command_count
-        ; proof_zkapp_command_count
-        }
+      ( match transaction_type with
+      | `Zkapp_command ->
+          Perf_histograms.add_span ~name:"snark_worker_zkapp_transition_time"
+            elapsed ;
+          Mina_metrics.(
+            Cryptography.(Counter.inc_one snark_work_zkapp_base_submissions))
+      | _ ->
+          Perf_histograms.add_span ~name:"snark_worker_nonzkapp_transition_time"
+            elapsed ;
+          Mina_metrics.(
+            Counter.inc Cryptography.snark_work_nonzkapp_base_time_sec
+              (Time.Span.to_sec elapsed)) ) ;
+      Base_snark_generated { transaction_type; elapsed }
 
-let emit_proof_metrics (result_with_spec : _ Partitioned_spec.Poly.t) =
+let emit_partitioned_metrics (result_with_spec : _ Partitioned_spec.Poly.t) =
   Mina_metrics.(Counter.inc_one Snark_work.completed_snark_work_received_rpc) ;
   match result_with_spec with
   | Partitioned_spec.Poly.Single { job; data; _ } ->
       emit_single_metrics ~single_spec:job.spec ~data
-  | Partitioned_spec.Poly.Sub_zkapp_command
+  | Sub_zkapp_command
       { job = { spec = Sub_zkapp_spec.Stable.Latest.Segment { spec; _ }; _ }
       ; data = Proof_carrying_data.{ data = elapsed; _ }
       ; _
       } ->
-      (* WARN: `Mina_metrics.snark_work_zkapp_base_submissions` for this branch
-         tracked inside the work partitioner, as there's no way to know it here.
-      *)
-      Perf_histograms.add_span ~name:"snark_worker_subzkapp_time" elapsed ;
```

Comment:
Why you decided to remove this `add_span` call?

Reply:
The intention is that's just a merged bucket of `snark_worker_sub_zkapp_command_segment_time`, and `snark_worker_sub_zkapp_command_merge_time`. 
I'd prefer to have them separated because they're different, kinda